packages/core/src/combinator/applicative.js

```diff
@@ -16,5 +17,9 @@ import { apply } from '@most/prelude'
  * @returns {Stream} stream containing all the applications of fs to xs
  */
 export function ap (fs, xs) {
+  if (containsCanonicalEmpty([fs, xs])) {
```

Comment:
Clever 😄

Do you feel like this communicates intent more clearly than `if(isCanonicalEmpty(fs) || isCanonicalEmpty(xs)) {`?

Reply:
Hmmm, now that I think about it, [`combine` was recently updated to detect canonical empty](https://github.com/mostjs/core/blob/master/packages/core/src/combinator/combine.js#L27).  We could potentially remove this check, but then _keep_ the new unit tests.

Thoughts?

Reply:
Oh, yes. We should definitely remove this check if the tests pass without it.

Reply:
Sorry I missed that in earlier reviews!